contents/dataVis.md

```diff
@@ -0,0 +1,390 @@
+make more sense. So, let's  `append` another `plot object`, a mutating one `!`, and
+add the corresponding legends by calling `axislegend`. This will collect all the `labels`
+you might have passed to your plotting functions. 
+
+```jl
+s = """
+    lines(1:10, (1:10).^2, label = "x²", linewidth = 2, linestyle = nothing,
+        figure = (; figure_padding = 5, resolution = (600,400), 
+            backgroundcolor = :grey90, fontsize = 16, font = "sans"), 
+        axis = (; xlabel = "x", title = "title", backgroundcolor = :white, 
+            xgridstyle=:dash, ygridstyle=:dash))
+    scatterlines!(1:10, (10:-1:1).^2, label = "Reverse(x)²")
+    axislegend("legend", position = :ct)
+    current_figure()
+    filename = "customPlotLeg" # hide 
+    Options(current_figure(); filename, caption="custom plot legend", label="customplotlegend") # hide
+    """
+sco(s)
+```
+
+However, having to writte this so much code just for two lines can become cumbersome
+and tired, so if you plan on doing a lot of plots with the same general aesthetics
+then setting a theme will be better. We can do this with `set_theme!()` as the following
+example illustrates, not a particular good set of attributes but you get the idea. 
+
+```jl
+s = """
+    set_theme!(resolution = (600,400), backgroundcolor = (:orange, 0.5), 
+        fontsize = 16, font = "sans",
+        Axis = (backgroundcolor = :white, xgridstyle=:dash, ygridstyle=:dash),
+        Legend = (bgcolor = (:red,0.2), framecolor = :dodgerblue)
+        )
+
+    lines(1:10, (1:10).^2, label = "x²", linewidth = 2, linestyle = nothing,
+        axis = (; xlabel = "x", title = "title"))
+    scatterlines!(1:10, (10:-1:1).^2, label = "Reverse(x)²")
+    axislegend("legend", position = :ct)
+    current_figure()
+    set_theme!() # in order to go back to the default settings. 
+    filename = "setTheme" # hide 
+    Options(current_figure(); filename, caption="set theme", label="setTheme") # hide
+    """
+sco(s)
+```
+
+For more on `themes` please go to section xxx (Refs)
+
+Before moving on into the next section, it's worthwhile to see an example 
+where an `array` of attributes are passed at once to a plotting function. 
+For this example we will use the `scatter` plotting function to do a bubble plot. 
+
+```jl
+s = """
+    let
+        Random.seed!(123)
+        n = 100
+        fig, ax, pltobj = CairoMakie.scatter(randn(n), randn(n), color = randn(n), 
+            label = "Bubbles",colormap = :plasma, markersize = 25*rand(n), 
+            figure = (; resolution = (550,400)), axis = (; aspect= DataAspect()))
+        limits!(-3, 3, -3, 3)
+        Legend(fig[1,2], ax, valign = :top)
+        Colorbar(fig[1,2], pltobj, height = Relative(3/4))
+        fig
+        Options(current_figure(); filename ="bubbleplot", caption="Bubble plot", label="bubbleplot") # hide 
+    end
+    """
+sco(s)
+```
+
+Where we have decomposed the tuple `FigureAxisPlot` into `fig, ax, pltobj`, in order to 
```

Comment:
I think it's best if we, in the Markdown, stick to one sentence per line. That way, the diff remains cleanest. Otherwise, rewrites will cause a big mess. But, I'm open to discussion about this and also this doesn't have to be fixed now.